compiler/rustc_interface/src/passes.rs

```diff
@@ -617,49 +625,79 @@ fn write_out_deps(
     let result = (|| -> io::Result<()> {
         // Build a list of files used to compile the output and
         // write Makefile-compatible dependency rules
-        let mut files: Vec<String> = sess
-            .source_map()
-            .files()
+        let source_files = sess.source_map().files();
+        let mut files: Vec<(String, Option<FileHash>)> = source_files
             .iter()
             .filter(|fmap| fmap.is_real_file())
             .filter(|fmap| !fmap.is_imported())
-            .map(|fmap| escape_dep_filename(&fmap.unmapped_path.as_ref().unwrap_or(&fmap.name)))
+            .map(|fmap| {
+                (
+                    escape_dep_filename(&fmap.unmapped_path.as_ref().unwrap_or(&fmap.name)),
+                    Some(FileHash::SourceFile(fmap.clone())),
+                )
+            })
             .collect();
 
         if let Some(ref backend) = sess.opts.debugging_opts.codegen_backend {
-            files.push(backend.to_string());
+            files.push((backend.to_string(), None)); // TO DO: is this something we need to hash?
```

Comment:
Yes. It is a dylib, so it contains an SVH. You could add a parameter with the result of `codegen_backend.metadata_loader()` to this function to read it, but that looks like a bit of a hack.